packages/firebase_auth/firebase_auth/ios/Classes/FLTFirebaseAuthPlugin.m

```diff
@@ -936,6 +892,113 @@ - (void)userGetIdToken:(id)arguments withMethodCallResult:(FLTFirebaseMethodCall
+    [requestedScopes addObject:ASAuthorizationScopeEmail];
+  }
+  request.requestedScopes = [requestedScopes copy];
+  request.nonce = [object stringBySha256HashingString:nonce];
+
+  ASAuthorizationController *authorizationController =
+      [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[ request ]];
+  authorizationController.delegate = object;
+  authorizationController.presentationContextProvider = object;
+  [authorizationController performRequests];
+}
+
+static void handleAppleAuthResult(FLTFirebaseAuthPlugin *object, id arguments, FIRAuth *auth,
+                                  FIRAuthCredential *credentials, NSError *error,
+                                  FLTFirebaseMethodCallResult *result) {
+  if (error) {
+    if (error.code == FIRAuthErrorCodeSecondFactorRequired) {
+      [object handleMultiFactorError:arguments withResult:result withError:error];
+    } else {
+      result.error(nil, nil, nil, error);
+    }
+    return;
+  }
+  if (credentials) {
+    [auth signInWithCredential:credentials
+                    completion:^(FIRAuthDataResult *authResult, NSError *error) {
+                      if (error != nil) {
+                        NSDictionary *userInfo = [error userInfo];
+                        NSError *underlyingError = [userInfo objectForKey:NSUnderlyingErrorKey];
+
+                        NSDictionary *firebaseDictionary =
+                            underlyingError.userInfo[@"FIRAuthErrorUserInfoDes"
+                                                     @"erializedResponseKey"];
+
+                        if (firebaseDictionary != nil && firebaseDictionary[@"message"] != nil) {
+                          // error from firebase-ios-sdk is
+                          // buried in underlying error.
+                          result.error(nil, firebaseDictionary[@"message"], nil, nil);
+                        } else {
+                          result.error(nil, nil, nil, error);
+                        }
+                      } else {
+                        result.success(authResult);
+                      }
+                    }];
+  }
+}
+
+- (void)userLinkWithProvider:(id)arguments
+        withMethodCallResult:(FLTFirebaseMethodCallResult *)result {
+  FIRAuth *auth = [self getFIRAuthFromArguments:arguments];
+
+  FIRUser *currentUser = auth.currentUser;
+  if (currentUser == nil) {
+    result.error(kErrCodeNoCurrentUser, kErrMsgNoCurrentUser, nil, nil);
+    return;
+  }
+
+  if ([arguments[@"signInProvider"] isEqualToString:kSignInMethodApple]) {
+    if (@available(iOS 13.0, macOS 10.15, *)) {
+      self.linkWithAppleUser = currentUser;
+      launchAppleSignInRequest(self, arguments, result);
+    } else {
+      NSLog(@"Sign in with Apple was introduced in iOS 13, update your Podfile with platform :ios, "
+            @"'13.0'");
+    }
+    return;
+  }
+#if TARGET_OS_OSX
+  NSLog(@"The Firebase Phone Authentication provider is not supported on the "
```

Comment:
I'm confused by this if macro since I don't see anything specific to phone auth in the else block? Else seems like Aby other provider rather than just phone auth?

Reply:
Nice catch, I forgot to change this line